lib/session.go

```diff
@@ -31,9 +31,36 @@ func (e *Session) Assume(arn string) (*Session, error) {
 	}
 
 	duration := expiration.Sub(time.Now())
-	creds, err := e.AWSCreds.AssumeRole(arn, duration)
+
+	var creds *AWSCredentials
+
+	arnStruct, err := ParseARN(arn)
+
 	if err != nil {
-		return nil, err
+		identityArn, err := e.AWSCreds.GetCallerIdentity()
+		if err != nil {
+			return nil, err
+		}
+
+		roleArn := &ARN{
+			Partition: identityArn.Partition,
+			Service:   "iam",
+			Region:    identityArn.Region,
+			AccountId: identityArn.AccountId,
+			Resource:  "role/" + arn,
+		}
+
+		roleArnString := roleArn.String()
+		fmt.Println("assumed ARN is :", roleArnString)
+		creds, err = e.AWSCreds.AssumeRole(roleArnString, duration)
+		if err != nil {
+			return nil, fmt.Errorf("Error assuming role '%s' which was interpreted as '%s'", arn, roleArnString)
```

Comment:
We should probably include the API error as we were before. There could be several reasons why the role assumption failed, we don't want to leave the user in the dark as to what the reason was.